src/harness/security_testcase.py

```diff
@@ -171,6 +172,29 @@ def doTestCipher(self, cipher, client_cert=None, client_key=None):
     with CiphersOverload(self._sas, [cipher], client_cert, client_key):
       self.assertRegistered([device_a])
 
+  def doSasTestCipher(self, cipher, client_cert, client_key, client_url):
+    """Does a cipher test as described in SSS tests 1 to 5 specification.
+
+    Args:
+      cipher: the cipher openSSL string name to test.
+      client_cert: SAS client certificate file in PEM format to use.
+      client_key: associated key file in PEM format to use.
+      client_url: base URL of the peer client SAS.
+    """
+
+    # Using pyOpenSSL low level API, does the SAS UUT server TLS session checks.
+    self.assertTlsHandshakeSucceed(self._sas_admin._base_url, [cipher],
+                                   client_cert, client_key)
+
+    # Does a regular SAS registration
+    self.SasReset()
+    certificate_hash = getCertificateFingerprint(client_cert)
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                   'url': client_url})
+    self._sas_admin.TriggerFullActivityDump()
+    with CiphersOverload(self._sas, [cipher], client_cert, client_key):
+      self._sas.GetFullActivityDump(client_cert, client_key)
```

Comment:
I think we should ensure that the GetFullActivityDump is successful. If not it should assert error should be raised.

Reply:
You're right. But this is already done.
The GetFullActivityDump calls _SasRequest that calls_RequestGet, and inside this function there is an 
assert http_code == 200, http_code
This should be enough I think.

Reply:
ok, thanks.

Reply:
Norbert,

We think this issue still needs to be fixed in the SSS test cases

The TLS handshake and GetFullActivityDump is done on URLs ._sas_admin._base_url). GetFullActivityDump and _sas._base_url respectively. But he URL (client_url) that is injected using inject peer sas is completely different. 

The client url is set to https://fake.sas.url.not.used.org/v1.2 while the ._sas_admin._base_url and ._sas._base_url are based on the sas.cfg file and use localhost:9000

Please note that all the security test cases including the ones that we submitted would need the fix. I think the fix is to not use the global variable “SAS_URL” but to automatically construct the URL based on the sas.cfg.

Perhaps if other reviewers agree then we can do one patch and fix all the SSS test cases after everything is merged.